log/logger.go

```diff
@@ -60,7 +63,16 @@ func Log() *logrus.Logger {
 
 // PrefixedLog return the global logger with prefix
 func PrefixedLog(prefix string) *logrus.Entry {
-	return logger.WithField("prefix", prefix)
+	return logger.WithField(prefixField, prefix)
+}
+
+// WithPrefix adds the given prefix to the logger.
+func WithPrefix(logger *logrus.Entry, prefix string) *logrus.Entry {
+	if existingPrefix, ok := logger.Data[prefixField]; ok {
+		prefix = fmt.Sprintf("%s.%s", existingPrefix, prefix)
+	}
+
+	return logger.WithField(prefixField, prefix)
```

Comment:
I haven't looked into it, but this might break the dashboards or whatever else depends on the prefix being exactly the name of the resolver.